File: Numerical Integration/Monte Carlo Method/Monte Carlo Integration.cpp
#include <iostream>
#include <math.h>
#include <time.h>

using namespace std;

//The function we want to integrate
float func(float x)
{
    return exp(-2 * x);
}



int main()
{
    srand(time(NULL));//Makes sure answer is different each time
    
    //Initializing values
    int totalPoints = 10000000.0;
    int pointUnderCurve = 0.0;

    float length = 2.5;
    float height = 1.3;

    float area = length * height;

    float x = 0;
    float y = 0;


    for(int i = 0; i < totalPoints; i++)
    {
        float xRand = (float)rand() / RAND_MAX;//These are temporary variables that are between 0 and 1
        float yRand = (float)rand() / RAND_MAX;

        x = xRand * length;//Using the temporary variables we scale them to fit in the rectangle
        y = yRand * height;

        if (y < func(x))
        {
            pointUnderCurve++;
        }
    }

    float areaUnderCurve = ((double) pointUnderCurve / (double) totalPoints) * area;
    cout << "Area under curve: " << areaUnderCurve;
}

File: Numerical Integration/Monte Carlo Method/Monte Carlo Integration.py
# ---------------------------------------------------------
# Author: Saad Mehmood
# Date: September 2024
# Description: This code was developed by Saad Mehmood as part 
# of the project NumSolverToolkit.
# Any use or modification of this code should include proper
# attribution to the original author.
# ---------------------------------------------------------

#Instead of calling the whole library we only call the "uniform()" function that 
#gives us a floating point value between two given numbers, similaryly we only call the "exp()" function from numpy
from random import uniform 
from numpy import exp

#Function we want to integrate
def func(x):
    return exp(-2 * x)


length = 2
height = 1
totalPoints = 100
pointsUnderCurve = 0

for i in range(totalPoints):
    x = uniform(0,2)
    y = uniform(0,1)

    if y < func(x):
        pointsUnderCurve = pointsUnderCurve + 1
    
area = length * height

areaUnderCurve = area * (pointsUnderCurve / totalPoints)

print("Area under the curve: ", areaUnderCurve)
